File: app/src/test/java/org/metabrainz/mobile/MockLookupRepository.java
package org.metabrainz.mobile;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.google.gson.Gson;

import org.metabrainz.mobile.data.repository.LookupRepository;
import org.metabrainz.mobile.data.sources.api.entities.WikiSummary;
import org.metabrainz.mobile.util.Resource;
import org.metabrainz.mobile.util.SingleLiveEvent;

import static org.metabrainz.mobile.EntityTestUtils.loadResourceAsString;

public class MockLookupRepository extends LookupRepository {

    public MockLookupRepository() {
        super(null);
    }

    @Override
    public LiveData<Resource<String>> fetchData(String entity, String MBID, String params) {
        MutableLiveData<Resource<String>> data = new MutableLiveData<>();
        String response = loadResourceAsString(entity + "_lookup.json");
        Resource<String> resource = new Resource<>(Resource.Status.SUCCESS, response);
        data.setValue(resource);
        return data;
    }

    @Override
    public SingleLiveEvent<Resource<WikiSummary>> fetchWikiSummary(String string, int method) {
        SingleLiveEvent<Resource<WikiSummary>> wiki = new SingleLiveEvent<>();
        String response = loadResourceAsString("artist_wiki.json");
        WikiSummary summary = new Gson().fromJson(response, WikiSummary.class);
        Resource<WikiSummary> resource = new Resource<>(Resource.Status.SUCCESS, summary);
        wiki.setValue(resource);
        return wiki;
    }
}

File: app/src/test/java/org/metabrainz/mobile/MockLookupRepository.kt
package org.metabrainz.mobile

import com.google.gson.Gson
import org.metabrainz.mobile.EntityTestUtils.loadResourceAsString
import org.metabrainz.mobile.data.repository.LookupRepository
import org.metabrainz.mobile.data.sources.api.entities.CoverArt
import org.metabrainz.mobile.data.sources.api.entities.WikiSummary
import org.metabrainz.mobile.util.Resource
import org.metabrainz.mobile.util.Resource.Status.SUCCESS

class MockLookupRepository : LookupRepository {

    override suspend fun fetchData(entity: String, MBID: String, params: String): Resource<String> {
        return Resource(SUCCESS, loadResourceAsString(entity + "_lookup.json"))
    }

    override suspend fun fetchWikiSummary(string: String, method: Int): Resource<WikiSummary> {
        val response = loadResourceAsString("artist_wiki.json")
        val summary = Gson().fromJson(response, WikiSummary::class.java)
        return Resource(SUCCESS, summary)
    }

    override suspend fun fetchCoverArt(MBID: String): Resource<CoverArt> {
        TODO("Not yet implemented")
    }
}